src/pywifes/extract_spec.py

```diff
@@ -0,0 +1,516 @@
+    vmin, vmax = np.percentile(
+        collapsed_cube[border_width:-border_width, border_width:-border_width], (5, 95)
+    )
+    # Show collapsed image
+    plt.imshow(collapsed_cube, vmin=vmin, vmax=vmax)
+
+    # Color of the overlay area
+    cmap = mcolors.ListedColormap(["white"])
+    alpha = 0.5
+
+    for index, source_ap in enumerate(source_apertures):
+        ap_index = index + 1
+        # Plot a overlaped transparent area
+        mask = source_ap.to_mask(method="center").to_image(np.shape(collapsed_cube))
+        mask[mask == 0] = np.nan
+        plt.imshow(mask, alpha=alpha, cmap=cmap)
+        # Plot theoretical aperture contourngit
+        source_ap.plot(color="white", lw=0.8, ls="--")
+        # Plot the aperture number
+        plt.text(
+            source_ap.positions[0],
+            source_ap.positions[1],
+            ap_index,
+            color="white",
+            ha="center",
+            va="center",
+            fontsize=12,
+            path_effects=[withStroke(linewidth=2, foreground="black")],
+        )
+
+    if sky_aps is not None:
+        for sky_ap in sky_aps:
+            # Plot a overlaped transparent area
+            mask = sky_ap.to_mask(method="center").to_image(np.shape(collapsed_cube))
+            mask[mask == 0] = np.nan
+            plt.imshow(mask, alpha=alpha, cmap=cmap)
+            # Plot theoretical aperture contourn
+            sky_ap.plot(color="white", lw=0.8, ls="--")
+
+
+def read_cube_data(cube_path):
+    cube_data = {
+        "sci": None,
+        "sci_hdr": None,
+        "var": None,
+        "var_hdr": None,
+        "wcs": None,
+        "binning_x": None,
+        "binning_y": None,
+    }
+
+    if cube_path is not None:
+        sci, sci_hdr = fits.getdata(cube_path, 0, header=True)
+        var, var_hdr = fits.getdata(cube_path, 1, header=True)
+        wcs = WCS(sci_hdr).celestial
+        binning_x, binning_y = np.int_(fits.getheader(cube_path, 0)["CCDSUM"].split())
+        cube_data["sci"] = sci
+        cube_data["sci_hdr"] = sci_hdr
+        cube_data["var"] = var
+        cube_data["var_hdr"] = var_hdr
+        cube_data["wcs"] = wcs
+        cube_data["binning_x"] = binning_x
+        cube_data["binning_y"] = binning_y
+
+    return cube_data
+
+
+def detect_extract_and_save(
+    blue_cube_path=None,
+    red_cube_path=None,
+    output_dir=None,
+    r_arcsec=2,
+    border_width=3,
+    sky_sub=False,
+    check_plot=False,
+    plot_output="detected_apertures_plot.pdf",
+):
+    # reading the data from cubes
+    # Blue arm
+    blue_cube_data = read_cube_data(blue_cube_path)
+    blue_sci = blue_cube_data["sci"]
+    blue_sci_hdr = blue_cube_data["sci_hdr"]
+    blue_var = blue_cube_data["var"]
+    blue_var_hdr = blue_cube_data["var_hdr"]
+    blue_wcs = blue_cube_data["wcs"]
+    if blue_cube_data["sci"] is not None:
+        binning_x = blue_cube_data["binning_x"]
+        binning_y = blue_cube_data["binning_y"]
```

Comment:
This is better, though you could do the same as reduce_data.py and use the dictionary lookup inline instead of making the intermediate variables. But, I'd say not worth changing, huge improvement as-is!

Reply:
We still need to check with Chris and Ian if we need to consider different binning configuration for red and blue arm.  I'l keep as it is now but I'll used dictionary lookup then.